### DISCUSSION TITLE
Custom workload's job metadata properties

### DISCUSSION THREAD

**Discussion by pallar-ms** [microsoft/Microsoft-Fabric-workload-development-sample#60] [General]
For custom workload item's job executions details, it appears some basic metadata such as start/end time, instance id, invoke type, etc. is maintained with Fabric and returned in the monitoring hub accordingly and the custom workload can provide status of the job via GetItemJobInstanceStateAsync() impl. 
What is the guidance if we want to have custom job metadata with additional properties to be displayed - is there an API contract already defined to return custom properties such that the job metadata maintained within Fabric is updated with the same and can be pulled in the monitoring hub  along with the job details currently returned, or do we need to maintain job/pipeline metadata for such properties in our own store and have the FE additionally request from the workload BE APIs to return that?

**Comment by IdanGoldfarb1989** [comment]
Hi, 
The basic metadata for job run, such as start/end time, instance ID, and invoke type, is maintained by Fabric and automatically returned in the monitoring hub as you mentioned. For any custom job metadata beyond this, there isn't a predefined API contract within Fabric to store additional properties.

If you want to display custom metadata (like additional properties specific to your workload  item job), you'll need to handle and store that information within your workload using the job instance ID as a key. This information can then be displayed in the UI, either within the job details pane or through a dedicated workload UI. To fetch the custom metadata, your frontend would need to request it from the workload's backend API, as Fabric doesn't store or manage these custom properties directly.

You can read more about the job details pane in our[ monitoring hub documentation](https://learn.microsoft.com/en-us/fabric/workload-development-kit/monitoring-hub#job-details-pane)
